diff --git a/path_access.module b/path_access.module
index 8e5ea92..ded6660 100644
--- a/path_access.module
+++ b/path_access.module
@@ -84,9 +84,13 @@ function path_access_init() {
   else {
     $page_match = TRUE;
   }
-
-  // Check that the current page is not a protected page before blocking user.
-  if (!$page_match && !path_access_protected_pages($path)) {
+  
+  $path = drupal_strtolower(drupal_get_path_alias($_GET['q']));
+  
+  $path_exists = menu_valid_path(array('link_path' => $path));
+  
+  // Check that the current page is not a protected page and does exist before blocking user.
+  if (!$page_match && !path_access_protected_pages($path) && $path_exists) {
     drupal_access_denied();
     exit;
   }
